File: kotlin/snippets/Snippet-002.kt
package fr.o80.kata

data class Product(
    val type: Int,
    val price: Float
)

fun get(list: List<Product>): Int {
    val ret = mutableListOf<Product>()
    for (x in list) {
        if (x.type != 19) {
            continue
        }
        ret.add(x)
    }
    return ret.size
}

File: snippets/js/Snippet-002.js
class Product {
    constructor ({type, price} = {}) {
        this.type = type;
        this.price = price;
    }
}

/**
 * Get the products that are about food.
 */
function get(list) {
    let ret = []
    for (let x in list) {
        if (x.type !== 19) {
            continue;
        }
        ret.push(x)
    }
    return ret.size
}